src/include/sof/audio/codec_adapter/codec/generic.h

```diff
@@ -80,6 +80,19 @@ enum module_cfg_fragment_position {
 	MODULE_CFG_FRAGMENT_SINGLE,
 };
 
+/**
+ * \enum module_processing_mode
+ * MODULE_PROCESSING_NORMAL: Indicates that module is expected to apply its custom processing on
+ *			      the input signal
+ * MODULE_PROCESSING_BYPASS: Indicates that module is expected to skip custom processing on
+ *			      the input signal and act as a passthrough component
```

Comment:
just curious, what the reason leaving it to the wrapper would be vs having the parent code take the stream away?

Reply:
Are you saying, we can it in the codec_adapter copy() and skip calling process() API altogether? Its a valid quastion and doable too. @mwasko @lgirdwood what do you think about it?

Reply:
> Are you saying, we can it in the codec_adapter copy() and skip calling process() API altogether? Its a valid question and doable too. @mwasko @lgirdwood what do you think about it?

That should work for modules that has the same input/output format, but for example for up/down mixer you might need to do at least some basic processing to meet output format criteria (even in bypass mode). You can skip mixing channels data but copy 4ch input into 2ch output would definitely end up with a failure.

Reply:
agreed, if in_params != out_params it would not work